### ISSUE TITLE
zc390 cannot find file when source dir contains periods (.)

### ISSUE THREAD

**Issue by adelosa** [z390development/z390#518]
I'm currently working on streamlining the package build in GitHub actions.
This involves running z390 with folders that contain periods - the folders include the z390 version..

I discovered that the cobol commands (zc390) does not like having periods anywhere in the source file name except the extension.
My source filename directory includes the z390 version number which has periods in it (for example z390_v1.8.1-37-g78163d26)

I believe the issue is related to the following code in zc390.java - method init_zc390 .. line 311
```java
		index = wk_name.indexOf('.');
		if (index == -1){
			zc_file_name = wk_name + ".CBL";
			mlc_file_name = wk_name + ".MLC";
		} else {
			zc_file_name = wk_name;
			mlc_file_name = wk_name.substring(0,index) + ".MLC";
		}
```
I suspect `wk_name` is using the full name including directory rather than just the last part of the name. I would need to do some debugging to see this.
I do know that replacing the periods (.) with underscores (_) resolved the issue. 

Quick test - created folder with periods (dir.with.periods) and added zCobol/demo/HELLO.CBL
```
❯ cblclg dir.with.periods/HELLO.CBL
12:00:31 HELLO     ZC390 START USING z390 v1.8.1-26-gcaaabb01 ON J2SE 17.0.8 08/20/23
ZC390E abort zcobol file I/O error - java.io.FileNotFoundException: dir.with.periods/HELLO (No such file or directory) on line 0
12:00:31 HELLO     ZC390 ENDED   RC=16 SEC= 0 MEM(MB)= 17 IO=3
cblclg ERROR: zc390 rc=16; see console
```
Renamed folder to use underscores instead of periods (dir_with_periods)
```
❯ cblclg dir_with_periods/HELLO.CBL
12:03:34 HELLO     ZC390 START USING z390 v1.8.1-26-gcaaabb01 ON J2SE 17.0.8 08/20/23
12:03:34 HELLO     ZC390 ENDED   RC= 0 SEC= 0 MEM(MB)= 17 IO=2
12:03:35 HELLO     MZ390 START USING z390 v1.8.1-26-gcaaabb01 ON J2SE 17.0.8 08/20/23
12:03:35 HELLO     MZ390 ENDED   RC= 0 SEC= 0 MEM(MB)=127 IO=21512
12:03:36 HELLO     LZ390 START USING z390 v1.8.1-26-gcaaabb01 ON J2SE 17.0.8 08/20/23
12:03:36 HELLO     LZ390 ENDED   RC= 0 SEC= 0 MEM(MB)= 18 IO=76
12:03:36 HELLO     EZ390 START USING z390 v1.8.1-26-gcaaabb01 ON J2SE 17.0.8 08/20/23
Hello World
12:03:36 HELLO     EZ390 ENDED   RC= 0 SEC= 0 MEM(MB)= 23 IO=187 INS=41
```

**Comment by jyganci** [comment]
Try replacing

index = wk_name.indexOf('.');

with index = wk_name.lastIndexOf('.');

**Comment by jyganci** [comment]
Your assessment appears to be correct.

Worked up a potential quick fix. Tested it with "bash/runcbldemos" and "rt/bash/runcbltests". Then created a directory two levels up from the z390 directory whose name has a period in it, copied zcobol/demo/HELLO.CBL to the directory, and ran bash/cblclg "bash/cblclg ../../cobol.dir/HELLO". Everything works fine. Fix shown in context follows.

The current (bad) code is commented. New code begins one line below the line that follows the commented code. Fix is seven lines.
```
        // args[0] may be relative path beginning with ".."
        String wk_name = tz390.fix_file_separators(args[0]);
  /* #518
        index = wk_name.indexOf('.');
        if (index == -1){
            zc_file_name = wk_name + ".CBL";
            mlc_file_name = wk_name + ".MLC";
        } else {
            zc_file_name = wk_name;
            mlc_file_name = wk_name.substring(0,index) + ".MLC";
        }
  #518 */
        index = wk_name.lastIndexOf(File.separator);

        if (wk_name.substring(index+1).lastIndexOf('.') == -1) {  // #518
            zc_file_name = wk_name + ".CBL";                      // #518
            mlc_file_name = wk_name + ".MLC";                      // #518
        } else {                                               // #518
            zc_file_name = wk_name;                           // #518
            mlc_file_name = wk_name.substring(0,wk_name.lastIndexOf('.')) + ".MLC";  // #518
        }                                                    // #518

        if (index >= 0){
            lab_file_dir = wk_name.substring(0,index+1);
        } else {
            lab_file_dir = "";
        }
        lab_file_name = wk_name.substring(index+1) + "_ZC_LABELS.CPY";
```

**Comment by adelosa** [comment]
Fixed by PR #522
